engine/src/org/pentaho/di/trans/steps/mergerows/MergeRows.java

```diff
@@ -257,7 +257,7 @@ public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
    * @throws KettleRowException
    *           in case there is a compatibility error.
    */
-  protected void checkInputLayoutValid( RowMetaInterface referenceRowMeta, RowMetaInterface compareRowMeta ) throws KettleRowException {
+  static void checkInputLayoutValid( RowMetaInterface referenceRowMeta, RowMetaInterface compareRowMeta ) throws KettleRowException {
```

Comment:
Why change this to a static? It would be better if you just used "this" in the check method and change the method back to protected.